### DISCUSSION TITLE
聊天记录标题问题

### DISCUSSION THREAD

**Discussion by cgofun** [WongSaang/chatgpt-ui#156] [General]
![image](https://user-images.githubusercontent.com/7732886/232320224-751278c9-2a4f-424f-88dd-750d701bbb9f.png)
标题生成出错，所以都是未命名，这个问题如何解决？

**Comment by WongSaang** [comment]
你知道怎么打开后端的 DEBUT 模式看日志吗？

**Reply by cgofun** [reply]
不知道，指导一下

**Reply by WongSaang** [reply]
修改 docker-composer.yml 文件的:

```
backend-wsgi-server:
image: wongsaang/chatgpt-ui-wsgi-server:latest
environment:
    DEBUG: True
```

然后 docker-compose up

**Reply by cgofun** [reply]
docker-compose up这方式启动不行，登录不了9000/admin，会提示CSRF错误，没办法设置key，就没办法调试，这对我这个菜鸟太难了

**Reply by WongSaang** [reply]
运行这个命令
```
DEBUG=True APP_DOMAIN="域名:9000"  docker-compose up -d
```

**Reply by cgofun** [reply]
[BEGIN] 2023-4-17 22:07:40
opuser-backend-wsgi-server-1  | Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301.
opuser-backend-wsgi-server-1  | Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301.
opuser-backend-wsgi-server-1  | Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301.
opuser-backend-wsgi-server-1  | 192.168.0.3 - - [17/Apr/2023:14:07:44 +0000] "POST /api/conversation/ HTTP/1.0" 200 482 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
opuser-backend-web-server-1   | 192.168.0.4 - - [17/Apr/2023:14:07:44 +0000] "POST /api/conversation/ HTTP/1.1" 200 553 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36" "-"
opuser-backend-web-server-1   | 192.168.0.4 - - [17/Apr/2023:14:07:44 +0000] "POST /api/gen_title/ HTTP/1.1" 400 41 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36" "-"
opuser-backend-wsgi-server-1  | 192.168.0.3 - - [17/Apr/2023:14:07:44 +0000] "POST /api/gen_title/ HTTP/1.0" 400 41 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"

[END] 2023-4-17 22:07:53

debug信息就这些， 感觉看不出啥问题，就是api/gen_title/请求失败，跟浏览器调试看到的一样

**Reply by WongSaang** [reply]
抱歉，之前下载下来的 docker-compose.yml 没有从环境变量中读取 DEBUT，现在再运行一次 

下载 docker-compose.yml

```
curl -L "https://raw.githubusercontent.com/WongSaang/chatgpt-ui/main/docker-compose.yml" -o docker-compose.yml
```

启动服务
```
DEBUG=True APP_DOMAIN="域名:9000"  docker-compose up -d
```

**Reply by cgofun** [reply]
下载更新了，下面是debug信息：
（另外建议open_api_key_setting默认True比较方便）

opuser-backend-wsgi-server-1  | Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301.
opuser-backend-wsgi-server-1  | Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301.
opuser-backend-wsgi-server-1  | messages: {'messages': [{'role': 'system', 'content': 'You are a helpful assistant.'}, {'role': 'user', 'content': '我如何在Javascript中进行HTTP请求？'}], 'tokens': 29}
opuser-backend-wsgi-server-1  | Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301.
opuser-backend-wsgi-server-1  | 192.168.0.3 - - [18/Apr/2023:03:28:15 +0000] "POST /api/conversation/ HTTP/1.0" 200 11449 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
opuser-backend-web-server-1   | 192.168.0.4 - - [18/Apr/2023:03:28:16 +0000] "POST /api/conversation/ HTTP/1.1" 200 13032 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36" "-"
opuser-backend-wsgi-server-1  | Bad Request: /api/gen_title/
opuser-backend-wsgi-server-1  | 192.168.0.3 - - [18/Apr/2023:03:28:16 +0000] "POST /api/gen_title/ HTTP/1.0" 400 41 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
opuser-backend-web-server-1   | 192.168.0.4 - - [18/Apr/2023:03:28:16 +0000] "POST /api/gen_title/ HTTP/1.1" 400 41 "http://172.16.11.247/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36" "-"

[END] 2023-4-18 11:29:00

![image](https://user-images.githubusercontent.com/7732886/232663624-e3acb3be-2b35-4dd3-be3b-babd28b73cb4.png)

**Comment by WongSaang** [comment]
[image: image.png]
点下这里看看，奇了个怪

**Reply by WongSaang** [reply]
![image](https://user-images.githubusercontent.com/46235412/232677713-df911103-aac5-452f-82b3-4bb9bac0bcc9.png)

**Reply by cgofun** [reply]
key是在的，gen_title响应没了。我的key是打开open_api_key_setting=True，然后在client前端页面上填入的，就会出现这种情况。刚刚我测试了，后端设置openai_api_key是没有问题的，能正常生成title。

![](https://user-images.githubusercontent.com/7732886/232680996-82370017-3810-47b0-9cd8-64bfbf155505.png)

![](https://user-images.githubusercontent.com/7732886/232681159-4f0f12c6-4fd0-4926-9dbe-c481469f33e9.png)

**Comment by WongSaang** [comment]
我知道了，你这个 key 是从前端传的，但是 gen_title 接口没有传这个参数。这是个bug，我稍后修复。你可以先在后台添加一个 key
[image: image.png]

**Reply by WongSaang** [reply]
![image](https://user-images.githubusercontent.com/46235412/232682716-08b7bcaf-fabb-4879-abbd-7acdae81fa85.png)

**Reply by WongSaang** [reply]
@cgofun 最新版本已经修复这个问题，直接重新执行部署脚本就可以。

**Reply by cgofun** [reply]
谢谢，我已经更新了
